### DISCUSSION TITLE
How can I download an older version of Homebrew

### DISCUSSION THREAD

**Discussion by TheGalacticAce** [Homebrew/discussions#4096] [Getting started] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.18
ORIGIN: https://github.com/Homebrew/brew
HEAD: 106af035578d5f8716ae9510c1f9ad8c19f73a79
Last commit: 4 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: fd20b0538c0d7662722cea2600111f97044547b0
Core tap last commit: 69 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: octa-core 64-bit haswell
Clang: 12.0.0 build 1200
Git: 2.33.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 10.15.7-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A
```

### Output of `brew doctor`

```shell
Warning: You are using macOS 10.15.
We (and Apple) do not provide support for this old version.
It is expected behaviour that some formulae will fail to build in this old version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from MacHomebrew on Twitter.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.


Warning: Broken symlinks were found. Remove them with `brew cleanup`:
  /usr/local/share/man/man1/brew-cask.1
  /usr/local/share/zsh/site-functions/_brew_cask
```

### Description of issue

How can I install an older version of Homebrew? According to the installation guide, I need macOS Big Sur (11.0) or higher and I have macOS Catalina (10.15.7).

**Comment by SMillerDev** [comment]
You can not, only the newest version of Homebrew is ever supported by the installer.

**Reply by TheGalacticAce** [reply]
Is there a way I can get it with a different version of the installer?

**Reply by SMillerDev** [reply]
Any commit of the installer will always install the latest version of Homebrew so that won't help.

**Comment by sroby** [comment, marked as answer]
Have you tried installing it anyway? Just because it's no longer supported, doesn't mean it won't work.

**Reply by TheGalacticAce** [reply]
It gives me an error. I can't say what it gives as I don't have my Mac with me at the moment

**Reply by TheGalacticAce** [reply]
The error says:
```
Error: Failed to link all completions, docs and manpages:
  Permission denied @ rb_file_s_symlink - (../../../Homebrew/completions/fish/brew.fish, /usr/local/share/fish/vendor_completions.d/brew.fish)
Failed during: /usr/local/bin/brew update --force --quiet
```

**Reply by gromgit** [reply]
> `Permission denied @ rb_file_s_symlink - (../../../Homebrew/completions/fish/brew.fish, /usr/local/share/fish/vendor_completions.d/brew.fish)`

That says Homebrew wasn't able to create the `/usr/local/share/fish/vendor_completions.d` symbolic link, so I'm guessing that `/usr/local/share/fish/vendor_completions.d` has incorrect permissions and/or ownerships.

What's the output of `/bin/ls -lde@ /usr/local/share/fish/vendor_completions.d`?

**Reply by TheGalacticAce** [reply]
```shell
drwxr-xr-x  3 pwalia  admin  96 Sep 14  2018 /usr/local/share/fish/vendor_completions.d
```

**Reply by gromgit** [reply]
Huh, that looks reasonable, assuming your username is `pwalia`. How about:
```
/bin/ls -lRae@ /usr/local/share/fish
```
?

**Reply by TheGalacticAce** [reply]
Sorry for the late response, here 
```shell
total 0
drwxr-xr-x  3 pwalia  admin   96 Sep 14  2018 .
drwxrwxr-x  9 pwalia  admin  288 Dec 21 21:30 ..
drwxr-xr-x  3 pwalia  admin   96 Sep 14  2018 vendor_completions.d

/usr/local/share/fish/vendor_completions.d:
total 0
drwxr-xr-x  3 pwalia  admin  96 Sep 14  2018 .
drwxr-xr-x  3 pwalia  admin  96 Sep 14  2018 ..
lrwxr-xr-x  1 pwalia  admin  77 Sep 14  2018 carthage.fish -> ../../../Cellar/carthage/0.30.1/share/fish/vendor_completions.d/carthage.fish
```

**Reply by gromgit** [reply]
Just to be sure, what's the output of `id -a`?

**Reply by TheGalacticAce** [reply]
uid=503(chinchilla-account) gid=20(staff) groups=20(staff),12(everyone),61(localaccounts),79(_appserverusr),80(admin),81(_appserveradm),98(_lpadmin),701(com.apple.sharepoint.group.1),702(com.apple.sharepoint.group.2),223(com.apple.access_loginwindow),703(com.apple.sharepoint.group.3),33(_appstore),100(_lpoperator),204(_developer),250(_analyticsusers),395(com.apple.access_ftp),101(com.apple.access_screensharing-disabled),399(com.apple.access_ssh),400(com.apple.access_remote_ae)

**Reply by gromgit** [reply]
>```
>uid=503(chinchilla-account) 
>```

There's your problem: the directory is writable only by its owner (`pwalia`), but you're logged in as `chinchilla-account`. There are ways to make it writable to `chinchilla-account`, but the resulting mish-mash of file ownerships creates its own headaches down the line. Better to log in as `pwalia` to do your Homebrew stuff.

**Reply by TheGalacticAce** [reply]
Thank you @gromgit , that worked!

**Comment by TheGalacticAce** [comment]
So I just tried `brew doctor` and turns out that I'm not allowed to talk about this so I guess that's all lol, thanks for the help

**Reply by gromgit** [reply]
If you mean this:
```
Warning: You are using macOS 10.15.
We (and Apple) do not provide support for this old version.
It is expected behaviour that some formulae will fail to build in this old version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from MacHomebrew on Twitter.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
```
That last line should be interpreted as "you're unlikely to receive a response **from the maintainers**, because they've more than enough on their plates".

The Homebrew community is more than its maintainers, though. There are quite a few people (including myself) happy to help out, as long as it's understood that base Homebrew will [always be moving forward](https://www.youtube.com/watch?v=0nyQAUcTrYw).

**Reply by TheGalacticAce** [reply]
Ah alright, I just unmarked it

**Reply by JacksonUptain** [reply]
I believe you can download an older version by going to the menu at the top and looking for the history of the website(e.g: when it was published) just by clicking on the hyper text you should be able to get the older version, you can also put the old npm tag

**Reply by euguk007** [reply]
Homebrew is dead for anything below Big Sur.

**Reply by gromgit** [reply]
> Homebrew is dead for anything below Big Sur.

Not quite. You won't have _pre-built bottles_ for most formulae below Big Sur, but almost all of those should still build from source in Catalina. Of course, that "can't build" set grows with each older version, but in general, if a formula fails to build in Homebrew, it likely won't even build manually on your older macOS anyway.

**Reply by volo-droid** [reply]
> Not quite. You won't have _pre-built bottles_ for most formulae below Big Sur

Just for my understanding, would the bottles that were created when 10.15 was supported still be available? I assume they're not pro-actively deleted from [GitHub Packages](https://github.com/orgs/Homebrew/packages), are they?

**Reply by osalbahr** [reply]
@volo-droid are you having issues with the current bottles? I don’t know if Apple made breaking changes to how the executables work since then.

**Reply by ZhongRuoyu** [reply]
> Just for my understanding, would the bottles that were created when 10.15 was supported still be available?

Yes, they are still available on GitHub Packages. You may install them directly, or for some formulae, by checking out the specific commit where they were once available (knowing that this is *not* supported).
